src/compiler/checker.ts

```diff
@@ -8337,6 +8352,29 @@ namespace ts {
                 !!(type.flags & TypeFlags.Intersection) && every((<IntersectionType>type).types, isValidBaseType);
         }
 
+        function resolveImplementsTypesOfClass(type: InterfaceType) {
+            type.resolvedImplementsTypes = type.resolvedImplementsTypes || emptyArray;
+            for (const declaration of type.symbol.declarations) {
+
+                if (!isClassLike(declaration)) continue;

```

Comment:
Is there a test of either fail case? I didn’t see one.

Reply:
The `isClassLike` test guards if there are multiple merged declarations, but in all fairness I can't think of way to trigger this. Since we are in a JS file this will always be a class. Only interfaces and namespaces would merge with the class (and be present in the `declarations` array) but since we don't support declaring these in JS files, there is no way to trigger it. (Merging with an interface from `ts` file would not be included in this `declarations` as this is just the local file symbol)

**Should I just remove the test ?** It is a more of a future proof, if `@interface` or `@namespace` are ever included, but it might be just best to let whoever implements that deal with this case.

As for the `!implementsTypeNodes` tests, while I did not add any new tests for it, there are existing tests for emitting class definitions from a js file, which all do not have any `@implements` so those tests cover this as well (ex: any class in `jsDeclarationsClasses.ts` would hit this condition)

Reply:
I think you should remove this check too. In future, if constructor functions support `@implements`, I'd prefer to just update `getEffectiveImplementsTypeNodes` and not here too.